### PR TITLE
Update src/library/Text

### PR DESCRIPTION
**Pull request by tx-steven**
Add custom props to text:
`altFont` - changes font to "Proxima Nova Semi Bold"
`bold (fontWeight: bold)`
`fontFamily`
`fontSize`
`lineHeight`

Sneak in custom props to Box
`borderBottom`
`borderTop`

### CODE REVIEW

**Review by tx-michael — APPROVED** [review]

**Review by tx-trent — APPROVED** [review]

File: src/library/Box/styled.js [hunk 1]
```diff
@@ -85,23 +85,28 @@ export const BoxRoot = styled('div', {
   };
 
   return {
-    ...(backgroundColor && { backgroundColor }),
-    ...(overflow && { overflow }),
-    ...(position && { position }),
-    ...(scrollable && { overflow: 'scroll' }),
+    /* Built-In Styles */
```
Comment (tx-nestor) [inline src/library/Box/styled.js]: I'd consider this an unnecessary comment, just adds noise to the diff.

File: src/library/Box/styled.js [hunk 2]
```diff
@@ -85,23 +85,28 @@ export const BoxRoot = styled('div', {
...
     ...componentStyleReset(theme),
     ...getResponsiveStyles({
       breakpoints,
       mapValueToProperty,
       styles: {
-        borderRadius,
+        borderRadius, // TargetX Custom Style
```
Comment (tx-nestor) [inline src/library/Box/styled.js]: same comment as above.

File: src/library/Text/propTypes.js [hunk 3]
```diff
@@ -3,6 +3,13 @@ import { bool, node, number, oneOf, oneOfType, string } from 'prop-types';
 import { ALIGN, APPEARANCE, FONT_WEIGHT } from './constants';
 
 export const textPropTypes = {
+  /* TargetX Custom Props */
+  bold: bool,
```
Comment (tx-nestor) [inline src/library/Text/propTypes.js:8]: I'd consider `bold` an alias/shorthand for `fontWeight: bold`, so if we're going to do this, let's also support the `fontWeight` prop as a passthrough

File: src/library/Text/propTypes.js [hunk 4]
```diff
@@ -3,6 +3,13 @@ import { bool, node, number, oneOf, oneOfType, string } from 'prop-types';
...
+  fontSize: string,
+  lineHeight: string,
+  semiBold: bool,
```
Comment (tx-nestor) [inline src/library/Text/propTypes.js]: 🤢 see below

File: src/library/Text/styled.js [hunk 5]
```diff
@@ -47,14 +47,21 @@ export const TextRoot = styled('p', {
...
+    /* Built-In Styles */
     align,
     appearance: propAppearance,
     as,
     color,
     defaultAppearance,
     fontWeight,
     inherit,
-    noMargins,
+    // noMargins,
```
Comment (tx-nestor) [inline src/library/Text/styled.js]: Don't remove `noMargins`, just set it to `true` by default.

File: src/library/Text/styled.js [hunk 6]
```diff
@@ -121,7 +128,16 @@ export const TextRoot = styled('p', {
       ...getCommonStyles(as, theme, truncate),
       // 1 - Not normalized because we actually want `##em` as applied value
       // 2 - Must come after commonStyles
-      marginBottom: noMargins ? 0 : theme.Text_marginBottom
```
Comment (tx-nestor) [inline src/library/Text/styled.js:124]: 1. I wouldn't remove this line. Just set noMargins = true by default.
2. As a fallback, we can also set the Text_marginBottom on the theme to 0.

File: src/library/Text/styled.js [hunk 7]
```diff
@@ -121,7 +128,16 @@ export const TextRoot = styled('p', {
...
+
+      /* TargetX Custom Styles */
+
+      /* Disables default bottom margin on text. */
```
Comment (tx-nestor) [inline src/library/Text/styled.js]: See above. Don't need to comment this out. Just adds noise to the diff.

File: src/library/Text/styled.js [hunk 8]
```diff
@@ -121,7 +128,16 @@ export const TextRoot = styled('p', {
...
+      // marginBottom: noMargins ? 0 : theme.Text_marginBottom
+
+      ...(bold && { fontWeight: 'bold' }),
```
Comment (tx-nestor) [inline src/library/Text/styled.js:136]: `bold` is an alias/shorthand. Must support `fontWeight` passthrough too.

File: src/library/Text/styled.js [hunk 9]
```diff
@@ -121,7 +128,16 @@ export const TextRoot = styled('p', {
...
+      ...(fontSize && { fontSize }),
+      ...(lineHeight && { lineHeight }),
+      ...(semiBold && { fontFamily: 'Proxima Nova Semi Bold' })
```
Comment (tx-nestor) [inline src/library/Text/styled.js]: I'm not a fan of the `semiBold` prop -- it assumes we have a semi-bold compatible font available AND, worse yet, you're currently hard-coding a font family string "Proxima Nova Semi Bold". Instead, we should simply support `fontFamily` passthrough prop.

Alternatively, I'm open to adding the concept of an "alternative font" directly into the theme and then add an `altFont` prop. That would use the alternative font configured in the theme.

**Review by tx-nestor — CHANGES_REQUESTED** [review]
@tx-steven  Let's talk briefly about this tomorrow.

File: src/library/Text/styled.js [hunk 10]
```diff
@@ -47,14 +47,22 @@ export const TextRoot = styled('p', {
...
+    /* Built-In Styles */
     align,
     appearance: propAppearance,
     as,
     color,
     defaultAppearance,
     fontWeight,
     inherit,
-    noMargins,
+    noMargins = true,
```
Comment (tx-nestor) [inline src/library/Text/styled.js:65]: On second thought, let's just remove this default setting `= true` for now.  Unnecessary diff since we're going to set `Text_marginBottom = 0` anyway. Also, I think ultimately, we'll end up support `marginTop` and `marginBottom` at some point.

File: src/library/Text/types.js [hunk 11]
```diff
@@ -13,6 +13,14 @@ type Appearance = $Keys<typeof APPEARANCE>;
 type FontWeight = $Keys<typeof FONT_WEIGHT> | number;
 
 export type TextProps = {
+  /* TargetX Custom Props */
+  altFont?: string,
```
Comment (tx-nestor) [inline src/library/Text/types.js:17]: @tx-trent @tx-michael 

Poll: What do you think of this prop name `altFont`?

Its purpose is to support a secondary/alternative font family. e.g., Proxima Nova (as Primary) and Proxima Nova Semi Bold (Secondary/Alternative).

Other ones I've considered:
- `<Text alt .../>` (by itself)
- `<Text altFamily .../>`
- `<Text secondary .../>`

Thoughts?

**Review by tx-nestor — APPROVED** [review]
@tx-steven  I'm approving BUT:
1. requested one change
2. polling for feedback on prop name `altFont`

File: src/library/Text/types.js (hunk 11 above)
Comment (tx-michael) [reply src/library/Text/types.js:17]: I would go with `secondary` here.
